internal/signerverifier/dsse/dsse.go

```diff
@@ -3,22 +3,34 @@ package dsse
 import (
 	"context"
 	"encoding/base64"
+	"encoding/json"
 
-	"github.com/adityasaky/gittuf/internal/signerverifier"
-	"github.com/adityasaky/gittuf/internal/signerverifier/ed25519"
+	"github.com/adityasaky/gittuf/internal/signerverifier/common"
 	"github.com/secure-systems-lab/go-securesystemslib/dsse"
 )
 
-// SignEnvelope is an opinionated API to sign TUF metadata. It's opinionated
-// because it assumes the payload is Base 64 encoded, which is the expectation
-// for TUF metadata generated by gittuf.
-func SignEnvelope(ctx context.Context, envelope *dsse.Envelope, signingKeyContents []byte) (*dsse.Envelope, error) {
-	// TODO: pass off to generic LoadSigner
-	signer, err := ed25519.NewSignerVerifierFromSecureSystemsLibFormat(signingKeyContents)
+const PayloadType = "application/vnd.gittuf+json"
+
+// CreateEnvelope is an opinionated interface to create a DSSE envelope. It
+// accepts instances of tuf.RootMetadata, tuf.TargetsMetadata, etc. and marshals
+// the input prior to storing it as the envelope's payload.
+func CreateEnvelope(v any) (*dsse.Envelope, error) {
+	b, err := json.Marshal(v)
 	if err != nil {
 		return nil, err
 	}
 
+	return &dsse.Envelope{
+		Signatures:  []dsse.Signature{},
```

Comment:
Double checking this is intentional and don't want nil

Reply:
In future, when `policy-staging` is used, we may have unsigned envelopes for some time as root holders add their keys and so on, for example. So we don't want to sign right from the start and also for consistency, I think we should store unsigned envelopes rather than the payload itself.